homeassistant/components/verisure/alarm_control_panel.py

```diff
@@ -55,33 +55,47 @@ def unique_id(self) -> str:
         """Return the unique ID for this entity."""
         return self.coordinator.entry.data[CONF_GIID]
 
-    async def _async_set_arm_state(self, state: str, code: str | None = None) -> None:
+    async def _async_set_arm_state(self, state: str, command_data: dict) -> None:
         """Send set arm state command."""
         arm_state = await self.hass.async_add_executor_job(
-            self.coordinator.verisure.set_arm_state, code, state
+            self.coordinator.verisure.request, command_data
         )
         LOGGER.debug("Verisure set arm state %s", state)
-        transaction = {}
-        while "result" not in transaction:
+        result = None
+        while result is None:
             await asyncio.sleep(0.5)
             transaction = await self.hass.async_add_executor_job(
-                self.coordinator.verisure.get_arm_state_transaction,
-                arm_state["armStateChangeTransactionId"],
+                self.coordinator.verisure.request,
+                self.coordinator.verisure.poll_arm_state(
+                    list(arm_state["data"].values())[0], state
+                ),
+            )
+            result = (
+                transaction.get("data", {})
+                .get("installation", {})
+                .get("armStateChangePollResult", {})
+                .get("result", None)
```

Comment:
All these gets are so ugly... can't we improve that in the library?

Reply:
The library doesn't return any typed models. That would require a very large change on the library, that I am not the owner of.

Reply:
@frenck does this block the PR?

Reply:
I think we can't do much about it in that case 👍 

../Frenck